sdk/python/feast/registry.py

```diff
@@ -689,6 +758,16 @@ def get_user_metadata(
     cached_registry_proto_created: Optional[datetime] = None
     cached_registry_proto_ttl: timedelta
 
+    def __new__(
```

Comment:
it's not clear to me why this is necessary - can you explain in the PR description what the bug is / how it's being fixed?

also are there any tests that would catch the bugs?

Reply:
https://docs.python.org/3/reference/datamodel.html#object.__new__

Basically, `__new__` is called before `__init__`, and can be used to *change* the object that's being created by a call to a class constructor. So in our case, we are calling `Registry()`, but we want to return a `SqlRegistry` based on the config information. This implementation achieves that.

I can add a test for this specifically.